### PR TITLE
docs(README.md): add section on nominal vs structural class types

### PR DESCRIPTION
**Pull request by jedwards1211**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -298,6 +298,43 @@ class Test {};
...
+// No errors!
+```
+
+You can work around this with tricks like the following:
+```ts
+class Foo {
+    private __nominal: void;
+};
+class Bar {
+    private __nominal: void;
```
Comment (oriSomething) [inline README.md]: In TS 3.7, you should `declare private __nominal: void;` otherwise it's a field

Comment (oriSomething) [reply README.md]: More over the hack used with `enum` which is a nominal in TS

Comment (jedwards1211) [reply README.md]: okay cool, do you have a link to more about what you're talking about with enums?

**Comment by niieani** [comment]
Thanks @jedwards1211! Since I don't know enough here, let's wait for @oriSomething to give us a hint of where to go from here.

File: README.md (hunk 1 above)
Comment (jedwards1211) [reply README.md]: okay, updated

**Comment by niieani** [comment]
Thank you @jedwards1211 and @oriSomething!

**Pull request merged**